File: app/src/main/java/com/securechat/app/ui/screen/MediaPreviewScreen.kt
package com.securechat.app.ui.screen

import android.net.Uri
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.layout.imePadding
import androidx.compose.foundation.layout.navigationBarsPadding
import androidx.compose.foundation.layout.statusBarsPadding
import androidx.compose.foundation.pager.HorizontalPager
import androidx.compose.foundation.pager.rememberPagerState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.InsertDriveFile
import androidx.compose.material.icons.automirrored.filled.Send
import androidx.compose.material.icons.filled.Close
import androidx.compose.material.icons.filled.PlayArrow
import androidx.compose.material.icons.filled.Visibility
import androidx.compose.material.icons.filled.VisibilityOff
import androidx.compose.material3.FilledIconButton
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.IconButtonDefaults
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.material3.TextField
import androidx.compose.material3.TextFieldDefaults
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import coil.compose.AsyncImage
import coil.request.ImageRequest

/**
 * Medya/dosya onizleme ekrani.
 *
 * Kullanici dosya/fotograf/video sectikten sonra gonderim oncesi onizleme gosterir.
 * Resimler buyuk onizleme, diger dosyalar ikon + bilgi olarak gosterilir.
 * Birden fazla secimde yatay kaydirma (pager) ile gecis yapilir.
 *
 * @param items Onizlenecek medya listesi (URI + metadata)
 * @param onSend Kullanici "Gonder" e bastiginda cagirilir
 * @param onDismiss Kullanici geri/iptal butonuna bastiginda cagirilir
 */
@OptIn(androidx.compose.foundation.ExperimentalFoundationApi::class)
@Composable
fun MediaPreviewScreen(
    items: List<MediaPreviewItem>,
    onSend: (List<MediaPreviewItem>, String, Boolean) -> Unit,
    onDismiss: () -> Unit
) {
    var captionText by remember { mutableStateOf("") }
    var isViewOnce by remember { mutableStateOf(false) }
    val context = LocalContext.current

    Box(
        modifier = Modifier
            .fillMaxSize()
            .background(Color(0xFF0A0F18))
            .statusBarsPadding()
            .navigationBarsPadding()
    ) {
        Column(modifier = Modifier.fillMaxSize()) {
            // Ust bar — kapat butonu ve dosya sayisi
            Row(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(horizontal = 8.dp, vertical = 8.dp),
                verticalAlignment = Alignment.CenterVertically
            ) {
                IconButton(onClick = onDismiss) {
                    Icon(
                        Icons.Default.Close,
                        contentDescription = "Kapat",
                        tint = Color.White
                    )
                }
                if (items.size > 1) {
                    Text(
                        text = "${items.size} dosya seçildi",
                        color = Color(0xFF94A3B8),
                        fontSize = 14.sp
                    )
                }
            }

            // Onizleme alani — icerik tipine gore
            Box(
                modifier = Modifier
                    .weight(1f)
                    .fillMaxWidth(),
                contentAlignment = Alignment.Center
            ) {
                if (items.size == 1) {
                    // Tek dosya — dogrudan goster
                    MediaPreviewContent(item = items[0], context = context)
                } else {
                    // Birden fazla — pager ile kaydirma
                    val pagerState = rememberPagerState(pageCount = { items.size })
                    Column(
                        horizontalAlignment = Alignment.CenterHorizontally
                    ) {
                        HorizontalPager(
                            state = pagerState,
                            modifier = Modifier.weight(1f)
                        ) { page ->
                            MediaPreviewContent(item = items[page], context = context)
                        }

                        // Sayfa gostergesi (noktalar)
                        Row(
                            modifier = Modifier.padding(vertical = 12.dp),
                            horizontalArrangement = Arrangement.Center
                        ) {
                            repeat(items.size) { index ->
                                Box(
                                    modifier = Modifier
                                        .padding(horizontal = 3.dp)
                                        .size(if (index == pagerState.currentPage) 8.dp else 6.dp)
                                        .clip(CircleShape)
                                        .background(
                                            if (index == pagerState.currentPage) Color(0xFF0EA5E9)
                                            else Color(0xFF475569)
                                        )
                                )
                            }
                        }
                    }
                }
            }

            // Alt kisim — chat ekranindaki MessageInputBar ile birebir ayni gorunum.
            // Sol: ataşman placeholder (yok burada), Orta: caption alani, Sag: "1" view-once
            // toggle (WhatsApp tarzi), en sag: gonder dairesi.
            Row(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(horizontal = 8.dp, vertical = 6.dp)
                    .clip(RoundedCornerShape(28.dp))
                    .background(Color(0xFF1E293B))
                    .padding(start = 12.dp, end = 6.dp, top = 4.dp, bottom = 4.dp)
                    .imePadding(),
                verticalAlignment = Alignment.Bottom
            ) {
                // Caption alani — chat input gibi BasicTextField
                androidx.compose.foundation.text.BasicTextField(
                    value = captionText,
                    onValueChange = { captionText = it.take(1000) },
                    modifier = Modifier
                        .weight(1f)
                        .padding(vertical = 10.dp),
                    textStyle = androidx.compose.ui.text.TextStyle(
                        color = Color.White,
                        fontSize = 16.sp
                    ),
                    cursorBrush = androidx.compose.ui.graphics.SolidColor(Color(0xFF3E7BFA)),
                    maxLines = 4,
                    decorationBox = { inner ->
                        if (captionText.isEmpty()) {
                            Text(
                                "Açıklama ekle...",
                                color = Color(0xFF64748B),
                                fontSize = 16.sp
                            )
                        }
                        inner()
                    }
                )

                Spacer(modifier = Modifier.width(6.dp))

                // "1" view-once toggle — WhatsApp tarzi dairesel rozet, gonder butonunun solunda
                Box(
                    modifier = Modifier
                        .size(40.dp)
                        .clip(CircleShape)
                        .background(
                            if (isViewOnce) Color(0xFF3E7BFA).copy(alpha = 0.18f)
                            else Color.White.copy(alpha = 0.06f)
                        )
                        .border(
                            width = 1.5.dp,
                            color = if (isViewOnce) Color(0xFF3E7BFA) else Color.White.copy(alpha = 0.25f),
                            shape = CircleShape
                        )
                        .clickable { isViewOnce = !isViewOnce },
                    contentAlignment = Alignment.Center
                ) {
                    Text(
                        text = "1",
                        color = if (isViewOnce) Color(0xFF3E7BFA) else Color.White.copy(alpha = 0.7f),
                        fontSize = 16.sp,
                        fontWeight = FontWeight.Bold
                    )
                }

                Spacer(modifier = Modifier.width(6.dp))

                // Gonder butonu — chat ekranindaki ile birebir ayni
                Box(
                    modifier = Modifier
                        .size(40.dp)
                        .clip(CircleShape)
                        .background(Color(0xFF3E7BFA))
                        .clickable { onSend(items, captionText.trim(), isViewOnce) },
                    contentAlignment = Alignment.Center
                ) {
                    Icon(
                        Icons.AutoMirrored.Filled.Send,
                        contentDescription = "Gönder",
                        tint = Color.White,
                        modifier = Modifier.size(20.dp)
                    )
                }
            }
        }
    }
}

/**
 * Tek bir medya ogesinin onizleme icerigi.
 * Resimler buyuk onizleme, diger dosyalar ikon + ad/boyut bilgisi gosterir.
 */
@Composable
private fun MediaPreviewContent(
    item: MediaPreviewItem,
    context: android.content.Context
) {
    if (item.isImage) {
        // Resim onizlemesi
        AsyncImage(
            model = ImageRequest.Builder(context)
                .data(item.uri)
                .crossfade(true)
                .build(),
            contentDescription = item.fileName,
            modifier = Modifier
                .fillMaxSize()
                .padding(16.dp),
            contentScale = ContentScale.Fit
        )
    } else if (item.isVideo) {
        // Video: thumbnail + oynat ikonu
        Box(
            modifier = Modifier
                .fillMaxSize()
                .padding(16.dp),
            contentAlignment = Alignment.Center
        ) {
            AsyncImage(
                model = ImageRequest.Builder(context)
                    .data(item.uri)
                    .crossfade(true)
                    .build(),
                contentDescription = item.fileName,
                modifier = Modifier.fillMaxSize(),
                contentScale = ContentScale.Fit
            )
            // Video ikonu
            Box(
                modifier = Modifier
                    .size(64.dp)
                    .clip(CircleShape)
                    .background(Color.Black.copy(alpha = 0.6f)),
                contentAlignment = Alignment.Center
            ) {
                Icon(
                    imageVector = Icons.Default.PlayArrow,
                    contentDescription = null,
                    tint = Color.White,
                    modifier = Modifier.size(36.dp)
                )
            }
        }
    } else {
        // Dosya onizlemesi — ikon + bilgi
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .padding(32.dp),
            horizontalAlignment = Alignment.CenterHorizontally,
            verticalArrangement = Arrangement.Center
        ) {
            Box(
                modifier = Modifier
                    .size(96.dp)
                    .clip(RoundedCornerShape(20.dp))
                    .background(Color(0xFF1E293B)),
                contentAlignment = Alignment.Center
            ) {
                Icon(
                    Icons.AutoMirrored.Filled.InsertDriveFile,
                    contentDescription = null,
                    tint = Color(0xFF0EA5E9),
                    modifier = Modifier.size(48.dp)
                )
            }

            Spacer(modifier = Modifier.height(16.dp))

            Text(
                text = item.fileName,
                color = Color.White,
                fontSize = 18.sp,
                fontWeight = FontWeight.Medium,
                textAlign = TextAlign.Center,
                maxLines = 2,
                overflow = TextOverflow.Ellipsis
            )

            Spacer(modifier = Modifier.height(8.dp))

            Text(
                text = formatFileSize(item.fileSize),
                color = Color(0xFF64748B),
                fontSize = 14.sp
            )

            if (item.mimeType.isNotEmpty()) {
                Spacer(modifier = Modifier.height(4.dp))
                Text(
                    text = item.mimeType,
                    color = Color(0xFF475569),
                    fontSize = 12.sp
                )
            }
        }
    }
}

/**
 * Onizleme ekraninda gosterilecek medya ogesi.
 */
data class MediaPreviewItem(
    val uri: Uri,
    val fileName: String = "",
    val mimeType: String = "",
    val fileSize: Long = 0L
) {
    val isImage: Boolean get() = mimeType.startsWith("image/")
    val isVideo: Boolean get() = mimeType.startsWith("video/")
}

private fun formatFileSize(bytes: Long): String {
    return when {
        bytes < 1024 -> "$bytes B"
        bytes < 1024 * 1024 -> "${bytes / 1024} KB"
        bytes < 1024 * 1024 * 1024 -> String.format("%.1f MB", bytes / (1024.0 * 1024.0))
        else -> String.format("%.2f GB", bytes / (1024.0 * 1024.0 * 1024.0))
    }
}

File: flutter_securechat/lib/src/features/chat/media_preview_screen.dart
import 'package:flutter/material.dart';

import '../../l10n/l10n.dart';
import '../../media/media_attachment.dart';
import '../../widgets/local_image_view.dart';

class MediaSendRequest {
  const MediaSendRequest({
    required this.attachments,
    required this.caption,
    required this.isViewOnce,
  });

  final List<MediaAttachment> attachments;
  final String caption;
  final bool isViewOnce;
}

class MediaPreviewScreen extends StatefulWidget {
  const MediaPreviewScreen({super.key, required this.attachments});

  final List<MediaAttachment> attachments;

  @override
  State<MediaPreviewScreen> createState() => _MediaPreviewScreenState();
}

class _MediaPreviewScreenState extends State<MediaPreviewScreen> {
  final _caption = TextEditingController();
  final _page = PageController();
  var _currentPage = 0;
  var _viewOnce = false;

  @override
  void dispose() {
    _caption.dispose();
    _page.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: const Color(0xff0a0f18),
      appBar: AppBar(
        backgroundColor: Colors.transparent,
        foregroundColor: Colors.white,
        title: widget.attachments.length > 1
            ? Text(context.l10n.files_selected(widget.attachments.length))
            : null,
      ),
      body: SafeArea(
        top: false,
        child: Column(
          children: [
            Expanded(
              child: PageView.builder(
                controller: _page,
                itemCount: widget.attachments.length,
                onPageChanged: (value) => setState(() => _currentPage = value),
                itemBuilder: (context, index) =>
                    _PreviewContent(attachment: widget.attachments[index]),
              ),
            ),
            if (widget.attachments.length > 1)
              Padding(
                padding: const EdgeInsets.only(bottom: 12),
                child: Row(
                  mainAxisAlignment: MainAxisAlignment.center,
                  children: List.generate(
                    widget.attachments.length,
                    (index) => AnimatedContainer(
                      duration: const Duration(milliseconds: 180),
                      width: index == _currentPage ? 8 : 6,
                      height: index == _currentPage ? 8 : 6,
                      margin: const EdgeInsets.symmetric(horizontal: 3),
                      decoration: BoxDecoration(
                        shape: BoxShape.circle,
                        color: index == _currentPage
                            ? const Color(0xff0ea5e9)
                            : const Color(0xff475569),
                      ),
                    ),
                  ),
                ),
              ),
            Container(
              margin: const EdgeInsets.fromLTRB(8, 6, 8, 6),
              padding: const EdgeInsets.fromLTRB(12, 4, 6, 4),
              decoration: BoxDecoration(
                color: const Color(0xff1e293b),
                borderRadius: BorderRadius.circular(28),
              ),
              child: Row(
                crossAxisAlignment: CrossAxisAlignment.end,
                children: [
                  Expanded(
                    child: TextField(
                      controller: _caption,
                      maxLength: 1000,
                      minLines: 1,
                      maxLines: 4,
                      style: const TextStyle(color: Colors.white),
                      decoration: InputDecoration(
                        hintText: context.l10n.add_caption,
                        counterText: '',
                        border: InputBorder.none,
                      ),
                    ),
                  ),
                  const SizedBox(width: 6),
                  Tooltip(
                    message: context.l10n.view_once,
                    child: InkWell(
                      key: const Key('media-view-once'),
                      onTap: () => setState(() => _viewOnce = !_viewOnce),
                      customBorder: const CircleBorder(),
                      child: AnimatedContainer(
                        duration: const Duration(milliseconds: 180),
                        width: 40,
                        height: 40,
                        alignment: Alignment.center,
                        decoration: BoxDecoration(
                          shape: BoxShape.circle,
                          color: _viewOnce
                              ? const Color(0xff3e7bfa).withValues(alpha: 0.18)
                              : Colors.white.withValues(alpha: 0.06),
                          border: Border.all(
                            width: 1.5,
                            color: _viewOnce
                                ? const Color(0xff3e7bfa)
                                : Colors.white.withValues(alpha: 0.25),
                          ),
                        ),
                        child: Text(
                          '1',
                          style: TextStyle(
                            color: _viewOnce
                                ? const Color(0xff3e7bfa)
                                : Colors.white70,
                            fontWeight: FontWeight.bold,
                          ),
                        ),
                      ),
                    ),
                  ),
                  const SizedBox(width: 6),
                  IconButton.filled(
                    key: const Key('media-send'),
                    onPressed: () => Navigator.pop(
                      context,
                      MediaSendRequest(
                        attachments: widget.attachments,
                        caption: _caption.text.trim(),
                        isViewOnce: _viewOnce,
                      ),
                    ),
                    icon: const Icon(Icons.send),
                  ),
                ],
              ),
            ),
          ],
        ),
      ),
    );
  }
}

class _PreviewContent extends StatelessWidget {
  const _PreviewContent({required this.attachment});

  final MediaAttachment attachment;

  @override
  Widget build(BuildContext context) {
    if (attachment.isImage) {
      return Padding(
        padding: const EdgeInsets.all(16),
        child: LocalImageView(
          path: attachment.path,
          fit: BoxFit.contain,
          errorBuilder: (_, _, _) => _FileDetails(attachment: attachment),
        ),
      );
    }
    if (attachment.isVideo) {
      return Stack(
        alignment: Alignment.center,
        children: [
          _FileDetails(attachment: attachment),
          Container(
            width: 64,
            height: 64,
            decoration: BoxDecoration(
              shape: BoxShape.circle,
              color: Colors.black.withValues(alpha: 0.6),
            ),
            child: const Icon(Icons.play_arrow, color: Colors.white, size: 36),
          ),
        ],
      );
    }
    return _FileDetails(attachment: attachment);
  }
}

class _FileDetails extends StatelessWidget {
  const _FileDetails({required this.attachment});

  final MediaAttachment attachment;

  @override
  Widget build(BuildContext context) {
    return Center(
      child: Padding(
        padding: const EdgeInsets.all(32),
        child: Column(
          mainAxisSize: MainAxisSize.min,
          children: [
            Container(
              width: 96,
              height: 96,
              decoration: BoxDecoration(
                color: const Color(0xff1e293b),
                borderRadius: BorderRadius.circular(20),
              ),
              child: const Icon(
                Icons.insert_drive_file,
                color: Color(0xff0ea5e9),
                size: 48,
              ),
            ),
            const SizedBox(height: 16),
            Text(
              attachment.fileName,
              textAlign: TextAlign.center,
              maxLines: 2,
              overflow: TextOverflow.ellipsis,
              style: const TextStyle(
                color: Colors.white,
                fontSize: 18,
                fontWeight: FontWeight.w500,
              ),
            ),
            const SizedBox(height: 8),
            Text(
              formatMediaFileSize(attachment.fileSize),
              style: const TextStyle(color: Color(0xff64748b)),
            ),
            const SizedBox(height: 4),
            Text(
              attachment.mimeType,
              style: const TextStyle(color: Color(0xff475569), fontSize: 12),
            ),
          ],
        ),
      ),
    );
  }
}

String formatMediaFileSize(int bytes) {
  if (bytes < 1024) return '$bytes B';
  if (bytes < 1024 * 1024) return '${bytes ~/ 1024} KB';
  if (bytes < 1024 * 1024 * 1024) {
    return '${(bytes / (1024 * 1024)).toStringAsFixed(1)} MB';
  }
  return '${(bytes / (1024 * 1024 * 1024)).toStringAsFixed(2)} GB';
}
